### PR TITLE
make development possible on different os/arch 

### PR DESCRIPTION
**Pull request by rewantsoni**
The PR does the following:
1. Allow images to be built using both podman and docker
2. Update OPM version to 1.20.0
3. Run go-test as a sub-target instead of Docker step
4. Remove depricated linters and remove :Z (Ref: https://access.redhat.com/solutions/6985126)
5. Add --platform flag to target a specific arch/os
6. Move from sqlite based to file based catalog source
7. update setup-env test to the latest version

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -127,8 +127,8 @@ bundle: manifests kustomize operator-sdk yq ## Generate bundle manifests and met
 		--patch '[{"op": "replace", "path": "/spec/replaces", "value": "$(REPLACES)"}]'
 	$(KUSTOMIZE) build $(MANIFEST_PATH) | sed "s|STATUS_REPORTER_IMAGE_VALUE|$(IMG)|g" | awk '{print}'| \
 		$(OPERATOR_SDK) generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS) --extra-service-accounts="$$($(KUSTOMIZE) build $(MANIFEST_PATH) | $(YQ) 'select(.kind == "ServiceAccount") | .metadata.name' -N | paste -sd "," -)"
-	sed -i "s|packageName:.*|packageName: ${CSI_ADDONS_PACKAGE_NAME}|g" "config/metadata/dependencies.yaml"
-	sed -i "s|version:.*|version: "${CSI_ADDONS_PACKAGE_VERSION}"|g" "config/metadata/dependencies.yaml"
+	sed -i '' "s|packageName:.*|packageName: "${CSI_ADDONS_PACKAGE_NAME}"|g" "config/metadata/dependencies.yaml"
+	sed -i '' "s|version:.*|version: "${CSI_ADDONS_PACKAGE_VERSION}"|g" "config/metadata/dependencies.yaml"
```
Comment (leelavg) [inline Makefile:131]: `bundle` target is also in downstream, make sure this works for linux as well.

File: Dockerfile [hunk 2]
```diff
@@ -17,8 +17,6 @@ COPY controllers/ controllers/
 COPY config/ config/
 COPY pkg/ pkg/
 COPY service/ service/
-# Run tests and linting
-RUN make go-test
```
Comment (leelavg) [inline Dockerfile:21]: it'd better to add info to the commit msg why this has to be removed and moved to sub target?

File: hack/make-tools.mk [hunk 3]
```diff
@@ -32,3 +32,7 @@ operator-sdk: ## Download operator-sdk locally if necessary.
 OPM = $(BIN_DIR)/opm
 opm: ## Download opm locally if necessary.
 	@./hack/get-tool.sh $(OPM) https://github.com/operator-framework/operator-registry/releases/download/v1.20.0/$(GOOS)-$(GOARCH)-opm
+
+ENVTEST ?= $(BIN_DIR)/setup-envtest
+envtest: ## Download envtest-setup locally if necessary.
+	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@latest)
```
Comment (leelavg) [inline hack/make-tools.mk:38]: add new line.

File: Makefile (hunk 1 above)
Comment (rewantsoni) [reply Makefile]: It works after replacing it with yq

**Comment by leelavg** [comment]
@rewantsoni re-running catalog-build job is still failing, could you ptal?

**Comment by leelavg** [comment]
In the PRs where that Job is passing I see docker was being used
```
2024-04-02T07:56:03.0311035Z docker push localhost:5000/ocs-dev/ocs-client-operator-bundle:latest
2024-04-02T07:56:03.0556101Z The push refers to repository [localhost:5000/ocs-dev/ocs-client-operator-bundle]
2024-04-02T07:56:03.0600541Z 2a633eeac220: Preparing
2024-04-02T07:56:03.0601790Z 902922e1dcae: Preparing
2024-04-02T07:56:03.0602293Z e444f57ccc15: Preparing
2024-04-02T07:56:03.0800517Z 902922e1dcae: Pushed
2024-04-02T07:56:03.0805677Z 2a633eeac220: Pushed
2024-04-02T07:56:03.0834704Z e444f57ccc15: Pushed
2024-04-02T07:56:03.1006832Z latest: digest: sha256:a57a598dd6134d6330ca0194ec1d6b0d1a5c04249bd7c99cc2e9e266eb90fac3 size: 939
```
current PR
```
2024-04-03T03:13:29.0510314Z /usr/bin/podman push localhost:5000/ocs-dev/ocs-client-operator-bundle:latest
2024-04-03T03:13:29.1367136Z Getting image source signatures
2024-04-03T03:13:29.1756168Z Error: trying to reuse blob sha256:9d5b3afa09e709e7bc6053a8e4ac7135f607f34bb2ef0586afdfcbb149da06c6 at destination: pinging container registry localhost:5000: Get "https://localhost:5000/v2/": http: server gave HTTP response to HTTPS client
2024-04-03T03:13:29.1776377Z make: *** [Makefile:141: bundle-push] Error 125
2024-04-03T03:13:29.1795569Z ##[error]Process completed with exit code 2.
```

File: hack/make-project-vars.mk [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
+
+ENVTEST_K8S_VERSION?=1.26
+
+ifeq ($(IMAGE_BUILD_CMD),)
+IMAGE_BUILD_CMD := $(shell command -v podman || echo "")
+endif
+
+ifeq ($(IMAGE_BUILD_CMD),)
+IMAGE_BUILD_CMD := $(shell command -v docker || echo "")
+endif
```
Comment (leelavg) [inline hack/make-project-vars.mk:21]: pls reverse this to be same as existing workflow.

Comment (rewantsoni) [reply hack/make-project-vars.mk]: done

**Review by leelavg — APPROVED** [review]

**Pull request merged**